common/src/main/java/com/wynntils/features/user/StatusOverlayFeature.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.mojang.blaze3d.platform.Window;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.config.Config;
+import com.wynntils.core.config.ConfigHolder;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.features.overlays.Overlay;
+import com.wynntils.core.features.overlays.OverlayPosition;
+import com.wynntils.core.features.overlays.annotations.OverlayInfo;
+import com.wynntils.core.features.overlays.sizes.GuiScaledOverlaySize;
+import com.wynntils.core.managers.Model;
+import com.wynntils.gui.render.FontRenderer;
+import com.wynntils.gui.render.HorizontalAlignment;
+import com.wynntils.gui.render.TextRenderSetting;
+import com.wynntils.gui.render.TextRenderTask;
+import com.wynntils.gui.render.VerticalAlignment;
+import com.wynntils.mc.event.RenderEvent;
+import com.wynntils.wynn.event.StatusEffectsChangedEvent;
+import com.wynntils.wynn.model.TabModel;
+import java.util.List;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class StatusOverlayFeature extends UserFeature {
+    private List<TextRenderTask> renderCache = List.of();
+
+    @OverlayInfo(renderType = RenderEvent.ElementType.GUI)
+    public final StatusOverlay statusOverlay = new StatusOverlay();
+
+    @Override
+    public List<Class<? extends Model>> getModelDependencies() {
+        return List.of(TabModel.class);
+    }
+
+    @SubscribeEvent
+    public void onStatusChange(StatusEffectsChangedEvent event) {
+        recalculateRenderCache();
+    }
+
+    private void recalculateRenderCache() {
+        renderCache = TabModel.getTimers().stream()
+                .map(statusTimer -> new TextRenderTask(statusTimer.asString(), statusOverlay.getTextRenderSetting()))
+                .toList();
+    }
+
+    public class StatusOverlay extends Overlay {
+        @Config
+        public FontRenderer.TextShadow textShadow = FontRenderer.TextShadow.OUTLINE;
+
+        private TextRenderSetting textRenderSetting;
+
+        protected StatusOverlay() {
+            super(
+                    new OverlayPosition(
+                            55,
+                            0,
+                            VerticalAlignment.Top,
+                            HorizontalAlignment.Right,
+                            OverlayPosition.AnchorSection.TopRight),
+                    new GuiScaledOverlaySize(250, 110));
```

Comment:
Gui scale size is dynamic, which might pose a problem. However, there is likely no nice resolution to this, except for maybe limiting the size somehow

Reply:
Why is it a problem?

Reply:
If you have enough effects it might overflow the displayed overlay size in the overlay configuration gui

Reply:
Oh yeah. Well...

Reply:
Best for a separate pr, don't think an issue is worth this either.